build.ps1

```diff
@@ -264,14 +269,17 @@ if (!$SkipBuild)
         $dotnetRestoreArgs = "restore `"$repoPath\InstrumentationEngine.sln`" --configfile `"$configFile`""
         if ($ARM64)
         {
-            $dotnetRestoreArgs = "$dotnetRestoreArgs /p:IncludeARM64='True'"
+            $dotnetRestoreArgs = "$dotnetRestoreArgs /p:IncludeARM64=True"
         }
 
         Invoke-ExpressionHelper -Executable "dotnet" -Arguments $dotnetRestoreArgs -Activity 'dotnet Restore Solution'
 
         # NuGet restore disregards platform/configuration
         $nugetRestoreArgs = "restore `"$repoPath\NativeNugetRestore.sln`" -configfile `"$configFile`""
         Invoke-ExpressionHelper -Executable "nuget" -Arguments $nugetRestoreArgs -Activity 'nuget Restore Solution'
+
+        # Restore global variable
+        $env:NUGET_PACKAGES = $tempGlobalCache
```

Comment:
Since we're setting `$ErrorActionPreference = 'Stop'` in this script, if any of the above commands fial, such as `dotnet restore`, this will never run and the user's set environment variables will be altered outside of the script's scope.

It'd also conflict with multiple instances of this script running at the same time (even if in different file paths).

I'm not sure if this exists, but if powershell has a way of defining environment variables for child processes that'd be best. If not, consider using a `trap` statement instead to guarantee the user's environment variables are always restored.

Reply:
Good suggestion, from [https://devblogs.microsoft.com/powershell/traps-vs-trycatch/](https://devblogs.microsoft.com/powershell/traps-vs-trycatch/) I'll use a `try-finally` since that provides more granular control. Seems they do similar things.